### PR TITLE
feat: basic support for encoding

### PR DESCRIPTION
**Pull request by targos**
@lpatiny @stropitek PTAL.
This should allow to encode RGBA, RGB, GREY and GREYA with 8 or 16 bit depth.

### CODE REVIEW

File: src/PNGEncoder.js [hunk 1]
```diff
@@ -1,25 +1,119 @@
...
+
+        this.writeUint32(this._png.width);
+        this.writeUint32(this._png.height);
+        this.writeByte(this._png.bitDepth);
+        this.writeByte(this._png.colourType);
+        this.writeByte(0); // Compression method
+        this.writeByte(0); // Filter method
+        this.writeByte(0); // Interlace method
+
+        this.writeUint32(0);
```
Comment (stropitek) [inline src/PNGEncoder.js]: This is the CRC right? Is it spec-compliant not to calculate it?

Comment (targos) [inline src/PNGEncoder.js]: I don't know. The Windows image viewer accepts it.

File: __tests__/encode.js [hunk 2]
```diff
@@ -1,8 +1,46 @@
...
-        console.log(data);
+    it('simple RGBA', () => {
+        const dataArray = new Uint8Array([255, 255, 255, 255, 0, 0, 0, 255, 0, 0, 0, 255, 255, 255, 255, 255]);
+        const data = encode({
+            width: 2,
+            height: 2,
+            data: dataArray
+        });
+        expect(data).toBeInstanceOf(Uint8Array);
+        const decoded = decode(data);
```
Comment (targos) [inline __tests__/encode.js]: Do you have a better idea to check the result? Seems not ideal to me to use our own decoder

Comment (stropitek) [inline __tests__/encode.js]: libpng is supposed to have png checker cli tool. But I don't know how to install it. My installed fedora repositories could not find a package that provides this.
http://www.libpng.org/pub/png/apps/pngcheck.html

File: src/PNGEncoder.js (hunk 1 above)
Comment (targos) [inline src/PNGEncoder.js]: CRC is now calculated in the encoder

File: __tests__/encode.js (hunk 2 above)
Comment (targos) [inline __tests__/encode.js]: I decided to use [pngjs](https://github.com/lukeapage/pngjs). The library supports the full specification and checks the CRC.

**Comment by targos** [comment]
Updated. I think it's ready now

**Pull request merged**